### DISCUSSION TITLE
Verificare l'integrità e l'autenticità di home assistant

### DISCUSSION THREAD

**Discussion by PiuCurioso** [indomus/forum#1675] [HUB: Home Assistant]
Carissimi buongiorno ho letto questo articolo https://indomus.it/news/home-assistant-adotta-codenotary-per-lintegrita-e-la-sicurezza-del-proprio-codice/
 che parla di come gli sviluppatori di Home Assistant garantiscono l'integrità e l'autenticità, ma non mi e' chiaro di come possa l'utente verificare l'integrita di Home Assistant una volta scaricato, ricalcolando il Checksum, e fare il controllo della firma PGP del checksum e essere certo che Home Assistant non sia stato manomesso.Grazie mille buona giornata

**Comment by indomus** [comment]
> Carissimi buongiorno ho letto questo articolo https://indomus.it/news/home-assistant-adotta-codenotary-per-lintegrita-e-la-sicurezza-del-proprio-codice/ che parla di come gli sviluppatori di Home Assistant garantiscono l'integrità e l'autenticità, ma non mi e' chiaro di come possa l'utente verificare l'integrita di Home Assistant una volta scaricato, ricalcolando il Checksum, e fare il controllo della firma PGP del checksum e essere certo che Home Assistant non sia stato manomesso.Grazie mille buona giornata

Hai ragione, è un tema caldissimo e interessante. Prova a vedere [sul forum di HA](https://github.com/home-assistant/core/issues), chiedi, vedi cosa rispondono.

**Reply by PiuCurioso** [reply]
Grazie ho scritto questo il link 
https://community.home-assistant.io/t/checksums-and-pgp-signatures/417448/2
attendiamo una risposta
Grazie buona giornata

**Reply by snakuzzo** [reply]
Facci sapere

**Reply by PiuCurioso** [reply]
> Facci sapere

Certo Snakuzzo, rimaniamo coordinati sull'argomento 
Buon pomeriggio

**Reply by PiuCurioso** [reply]
Carissimi mi dispiace :( ma non ho avuto alcuna risposta.Peccato avevo messo su un piedistallo Home Assistant ritengo che questo sia un punto debole, e' vero che i più non si interessano a questi aspetti, però a mio modesto avviso é una grande mancanza non dare agli utenti la possibilità  di verificare l'autenticità di Home Assistant.Ma voi come comunita avete parlato con gli sviluppatori di queste cose? mi piacerebbe capire il perche di questa scelta.E voi che idea vi siete fatti? Grazie mille Buon pomeriggio

**Reply by indomus** [reply]
Studia [come funziona CodeNotary](https://codenotary.com/products/ci-cd/) con integrazione GitHub, ne capirai di più.
Per inciso: [tutte le immagini Docker di HA](https://hub.docker.com/r/homeassistant/home-assistant/tags) hanno a disposizione lo sha256, ricalcolatelo e vedi se torna.

**Reply by PiuCurioso** [reply]
OK grazie mille :)
Un caro saluto
